### PR TITLE
Provide a RulesStateInterface 2453533

### PR DESCRIPTION
**Pull request by p4trizio**
dear Wolfgang, hope you are going to read this: I've been struggling with git and rebasing branches and masters, eventually I had to create another branch called "-bis" 
I hope this is not going to create troubles in your normal workflow

### CODE REVIEW

File: src/Engine/RulesStateInterface.php [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+/**
+ * Defines an interface for the rules state.
+ */
+interface RulesStateInterface {
+
+  /**
+   * Adds the given variable to the given execution state.
+   *
+   * @param string $name
+   *   The varible name.
```
Comment (klausi) [inline src/Engine/RulesStateInterface.php]: should be "variable"

File: src/Engine/RulesStateInterface.php [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  /**
+   * Gets a variable.
+   *
+   * @param string $name
+   *   The name of the variable to return.
+   *
+   * @return \Drupal\Core\Plugin\Context\ContextInterface
+   *   The variable wrapped as context.
+   *
+   * @throws RulesEvaluationException
```
Comment (klausi) [inline src/Engine/RulesStateInterface.php]: Should use the fully qualified name space here.

File: src/Engine/RulesStateInterface.php [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+   *   The selector string, e.g. "node:uid:entity:mail:value".
+   * @param string $langcode
+   *   (optional) The language code used to get the argument value if the
+   *   argument value should be translated. Defaults to
+   *   LanguageInterface::LANGCODE_NOT_SPECIFIED.
+   *
+   * @return \Drupal\Core\TypedData\TypedDataInterface
+   *   The variable wrapped as typed data.
+   *
+   * @throws RulesEvaluationException
```
Comment (klausi) [inline src/Engine/RulesStateInterface.php]: Should use the fully qualified name space here.

Comment (p4trizio) [inline src/Engine/RulesStateInterface.php]: @klausi just made requested changes, thanks for feedback.
Should I "squash" commits into a single one?

**Comment by klausi** [comment]
Merged, thanks!

**Pull request closed without merging**